src/particle/converters/bimap.py

```diff
@@ -61,21 +81,22 @@ def __init__(self, class_A, class_B, converters=(int, int), filename=None):
         >>> bimap = BiMap(PDGID, PythiaID, filename=filename)
         """
 
-        self.class_A = class_A
-        self.class_B = class_B
+        self.class_A = class_A  # type: Type[A]
+        self.class_B = class_B  # type: Type[B]
 
         name_A = self.class_A.__name__.upper()
         name_B = self.class_B.__name__.upper()
 
         if filename is None:
             filename = "{a}_to_{b}.csv".format(a=name_A.lower(), b=name_B.lower())
-            filename = data.open_text(data, filename)
-        elif not hasattr(filename, "read"):
+            file_object = data.open_text(data, filename)
+        elif hasattr(filename, "read"):
+            file_object = filename
+        else:
             # Conversion to handle pathlib on Python < 3.6:
-            filename = str(filename)
-            filename = open(filename)
+            file_object = open(str(filename))
 
-        with filename as _f:
+        with file_object as _f:
```

Comment:
It's best to avoid changing types of a variable in the middle of a function. It also makes it easier for a reader to follow, generally.

Reply:
Good point. Thanks for the improvement.